File: spring-webflux/src/main/java/victor/training/reactive/usecase/complex/ExceptionInterceptor.java
package victor.training.reactive.usecase.complex;

import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@Slf4j
@RestControllerAdvice
class ExceptionInterceptor {

  @ExceptionHandler(Throwable.class)
  @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
  public String method(Throwable ex) {
    log.error("Intercepted exception: " + ex.getMessage(), ex);
    return "Bad luck: " + ex.getMessage();
  }
}

File: spring-webflux/src/main/java/victor/training/reactive/usecase/complex/ExceptionInterceptor.kt
package victor.training.reactive.usecase.complex

import org.slf4j.LoggerFactory
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class ExceptionInterceptor {
    private val log = LoggerFactory.getLogger(ExceptionInterceptor::class.java)

    @ExceptionHandler(Exception::class)
    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
    fun serverExceptionHandler(ex: Exception): String {
        log.error(ex.message, ex)
        return "Oups: " + ex.message
    }
}